README.md

```diff
@@ -1 +1,208 @@
+
+Return a recognized MIME type essence if the format of the content is identified else return a binary MIME type essence. The function is called by `xtractmime.extract_mime` when *Content-Type* headers are unknown (i.e. `content_types` argument is `None`)
+
+### function `xtractmime._sniff_mislabled_binary : Optional[bytes]`
+**Parameters:**
+
+* `input_bytes: bytes`
+
+Return a text or a binary MIME type essence. The function determines whether a binary resource has been mislabeled as plain text.
+The function is related to a bug in some older installations of *Apache* that causes the provided byte sequences to supply
+one of the *Content-Type* headers mentioned in [Section 5.1](https://mimesniff.spec.whatwg.org/commit-snapshots/609a3a3c935fbb805b46cf3d90768d695a1dcff2/#interpreting-the-resource-metadata) of MIME sniffing standards when serving files with unrecognized MIME types.
+
+### function `xtractmime._sniff_mislabled_feed : Optional[bytes]:`
+**Parameters:**
+
+* `input_bytes: bytes`
+* `supplied_type: bytes`
+
+Return a MIME type essence if the content is identified as a XML-based RSS feed or an Atom feed else return supplied MIME type. The function determines whether a feed has been mislabeled as HTML.
+
+### function `xtractmime.is_binary_data : bool`
+**Parameters:**
+
+* `input_bytes: bytes`
+
+Return *`True`* if the provided byte sequence contains any binary data bytes else *`False`*
+ 
+### MIME group functions
+
+These functions provide an additional functionality to determine whether a MIME type belongs to a specific MIME type group or not.
+Supported MIME type groups include *Image MIME type*, *Audio or Video MIME type*, *Font MIME type*, *ZIP-Based MIME type*, *Archive MIME type*, *XML MIME type*, *HTML MIME type*, *Scriptable MIME type* (includes *XML MIME type*, *HTML MIME type*, or any MIME type whose essence is "application/pdf"), *JavaScript MIME type*, *JSON MIME type*.
+
+**List of available functions:**
+
+* `xtractmime.mimegroups.is_image_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_audio_video_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_font_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_zip_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_archive_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_xml_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_html_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_scriptable_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_javascript_mime_type(mime_type: bytes) -> bool`
+* `xtractmime.mimegroups.is_json_mime_type(mime_type: bytes) -> bool`
+
+**Example**
+```python
+>>> from xtractmime.mimegroups import is_html_mime_type, is_image_mime_type, is_zip_mime_type
+>>> mime_type = b'text/html'
+>>> is_html_mime_type(mime_type)
+True
+>>> is_image_mime_type(mime_type)
+False
+>>> is_zip_mime_type(mime_type)
+False
+```
```

Comment:
````suggestion
### MIME type group functions

The following functions return `True` if a given MIME type belongs to a certain 
[MIME type group](https://mimesniff.spec.whatwg.org/#mime-type-groups), or 
`False` otherwise:

```
xtractmime.mimegroups.is_image_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_audio_video_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_font_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_zip_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_archive_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_xml_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_html_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_scriptable_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_javascript_mime_type(mime_type: bytes) -> bool
xtractmime.mimegroups.is_json_mime_type(mime_type: bytes) -> bool
```
````